File: src/simulation/agent/brainPackage/BehaviorManager.java
package simulation.agent.brainPackage;

import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

import simulation.common.Node;
import simulation.common.PolarCoordinate;
import simulation.common.State;


public class BehaviorManager {

private Behavior swarmBehavior;
private Behavior crashBehavior;
private Behavior currentBehavior;
protected List<Node> nodeList;
private List<PolarCoordinate> newPostions;

public BehaviorManager(Behavior swarmBehavior, Behavior crashBehavior){
	this.swarmBehavior = swarmBehavior;
	this.crashBehavior = crashBehavior;
	newPostions = new ArrayList<PolarCoordinate>();
}

public void setSwarmState(){
	currentBehavior = swarmBehavior;
}

public void setCrashedState(){
	currentBehavior = crashBehavior;
}

public State getCurrentState(){
	return currentBehavior.getNextState(newPostions, nodeList);
}

public void updateDeltaPostions(List <PolarCoordinate> polarCoordinates) throws IOException{
	newPostions = polarCoordinates;
}

public void updateNodeList(List<Node> nodeList){
	this.nodeList = nodeList;
}
}

File: src/simulation/agent/brainPackage/BehaviorManager.ts
import { Behavior } from './Behavior';
import { Node } from '../../common/Node';
import { PolarCoordinate } from '../../common/PolarCoordinate';
import { State } from '../../common/State';

export class BehaviorManager {
  private swarmBehavior: Behavior;
  private crashBehavior: Behavior;
  private currentBehavior: Behavior;
  protected nodeList: Node[] = [];
  private newPositions: PolarCoordinate[] = [];

  constructor(swarmBehavior: Behavior, crashBehavior: Behavior) {
    this.swarmBehavior = swarmBehavior;
    this.crashBehavior = crashBehavior;
    this.currentBehavior = swarmBehavior;
    this.newPositions = [];
  }

  setSwarmState(): void {
    this.currentBehavior = this.swarmBehavior;
  }

  setCrashedState(): void {
    this.currentBehavior = this.crashBehavior;
  }

  getCurrentState(): State {
    return this.currentBehavior.getNextState(this.newPositions, this.nodeList);
  }

  updateDeltaPositions(polarCoordinates: PolarCoordinate[]): void {
    this.newPositions = polarCoordinates;
  }

  updateNodeList(nodeList: Node[]): void {
    this.nodeList = nodeList;
  }
}
